### DISCUSSION TITLE
Organisation invite in Vaultwarden not working

### DISCUSSION THREAD

**Discussion by NeoLizzard** [dani-garcia/vaultwarden#3879] [Q&A]
### Subject of the issue

Registered user in Vaultwarden get organisation invite email after beeing add by organisation administrator, but after clicking the join organisation link from mail user status in the backend doesn't change, even though message on user side appear  saying "you successfully accept the organisation invitation. The status in the backend stay for the user at "invited" and there is no way to confirm that the invitation got accepted. Status doesn't change to pending or similar, as it did in the past.

### Deployment environment

* Install method: Docker image, Debian 12, docker-compose setup

### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.29.2
* Web-vault version: v2023.7.1
* OS/Arch: linux/x86_64
* Running within Docker: true (Base: Debian)
* Environment settings overridden: false
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Browser/Server Time Check: true
* Server/NTP Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: MySQL
* Database version: 11.1.2-MariaDB-1:11.1.2+maria~ubu2204
* Clients used: Windows 11 and Web
* Reverse proxy and version: nginx (latest)
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** 

```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": false,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_session_lifetime": 20,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "auth_request_purge_schedule": "30 * * * * *",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "*****://**********************************************************************************************",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://*************",
  "domain_origin": "*****://*************",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "*********",
  "invitations_allowed": false,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": "/data/vaultwarden.log",
  "log_level": "error",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "********************,*********************",
  "org_events_enabled": false,
  "org_groups_enabled": false,
  "password_hints_allowed": false,
  "password_iterations": 600000,
  "push_enabled": false,
  "push_installation_id": "***",
  "push_installation_key": "***",
  "push_relay_uri": "https://push.bitwarden.com",
  "reload_templates": false,
  "require_device_email": true,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sendmail_command": null,
  "sends_allowed": false,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": false,
  "signups_domains_whitelist": "******,********,*********",
  "signups_verify": true,
  "signups_verify_resend_limit": 3,
  "signups_verify_resend_time": 86400,
  "smtp_accept_invalid_certs": true,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": "plain",
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": null,
  "smtp_from": "******************",
  "smtp_from_name": "Vault Warden Passwortmanager",
  "smtp_host": "***********",
  "smtp_password": null,
  "smtp_port": 25,
  "smtp_security": "starttls",
  "smtp_ssl": null,
  "smtp_timeout": 60,
  "smtp_username": null,
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": 4,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_sendmail": false,
  "use_syslog": false,
  "user_attachment_limit": 1024000,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": true,
  "websocket_port": 3012,
  "yubico_client_id": "********",
  "yubico_secret_key": "***",
  "yubico_server": "https://api2.yubico.com"
}
```
</details>

### Steps to reproduce

Create Organisation
Add Domains in Administration interface for domains to be able to accept invites
Invite user to vaultwarden
Enable MFA for user account (yubikey)
Change Organisation settings policy to:
- enforce 2FA -> yes
- request strong master-password -> yes
- min settings for password-generator -> yes
- single organisation -> yes
- remove send -> yes
(rest default)

### Expected behaviour

User click invitation link for organisation, admin get notification, accept the pending request, user get access to organisation

### Actual behaviour

User click invitation link for organisation, get confirmation that his link invite got processed successfully - nothing else happens (user stay at invited state in backend from administrator perspective)

**Comment by BlackDex** [comment]
Have you followed https://bitwarden.com/help/managing-users/#onboard-users

Also the user need to create the account via the link with the token. Of they somehow didn't used that link, or revisited the page manually the invite will not be processed.

If a user successfully used the invite, the status of that user will change, after which you can confirm the user. The rest is basically all explained in the link above.

So if the state stays at invited, the user didn't used the link correctly.
They can click on the link again, login, and should receive a message they there invite is confirmed. This also should send a mail to the person who invited the user.

**Comment by NeoLizzard** [comment]
I followed the instructions from the link to the bone. (And in addition - it did work in the past aswell)
At some point we had a update for the docker container - and from that moment on it didn't work. But I can't tell which, since we didn't invite new users and just noticed recently that it stopped working. So it's hard to pinpoint when exactly the breaking change happend. Everything else (except for updating the docker container) did not get modified since, which make the list of possible reasons short.

In regards of the user - we did two test it also with a fresh new created user and a fresh installation (and a new organisation) - so everything from scratch - still same result. Something get stuck. And I can't figure out where or why. 

The "accept invitation" message appear (as expected) at user side - but the status get not changed and a email to admin get not send.
If there is any particular place you want me to get you the logs from - just tell me. I tried to figure it out myself for quiet some time. But I got now out of options.

**Comment by tinerop** [comment]
Same issue here. Using Vaultwarden Addon of Home assistant.

The invited person (a valid user) with a registered local vaultwarden account receives the email invitation in their mail inbox. Clicking the button "Join Organization Now" redirects to the home assistant login website. I login with Home Assitant username and password and i am on my homeassistant in the browser - and done. There is no message i would usually expect like "sucessfully added as user"... nothing. 
In the vaultwarden Admin page the invited users stays "invited" forever.

**Reply by BlackDex** [reply]
In your case i would suspect a wrong configured `DOMAIN` setting.

Check the admin interface.

**Reply by tinerop** [reply]
@BlackDex you have a response time like a fire department!  - This was the right hint. It works now as described in the user manual. Everything works fine including confirmations.... Thanks  a lot!

Vaultwarden Admin / General Settings / domain

I had the url of my home assistant instance inside.... i changed it to the vaultwarden webadmin page. - Now everything (Organisation invite in Vaultwarden not working )works perfect. Thanks!

**Reply by BlackDex** [reply]
Glad it works!

**Reply by SagarKandel** [reply]
no it doesnt work for me

**Reply by BlockheadVFX-IT** [reply]
So i created an organisation in Vaultwarden , and then sent an invite to another user. He gets the invitation link to Join organisation. Then he clicks on the link and gets prompted either to create an account or login. He then goes ahead and creates and account, which successfully creates his account. When he logs back in , he doesn't have access to the shared vault/organisation. 

I then resent invite and asked him to join, he then goes ahead and tried to join using Login method, because he already created an account in upper step.

I then as a creator of that organisation go and check , and its still in invited state , instead of being categorised as needs confirmation.

It has been such a frustrating process to try again and again . Can someone help? 

![image](https://github.com/dani-garcia/vaultwarden/assets/152861034/2edb0b18-98d8-4281-8378-3e89a4d8472b)

**Reply by BlockheadVFX-IT** [reply]
Sorry hiding things as per company's stricter sec requirements 
I did trying adding email domain whitelist , but i had same results 

![image](https://github.com/dani-garcia/vaultwarden/assets/152861034/e76ea46f-270a-4696-90be-5f1b5207bd0b)
![image](https://github.com/dani-garcia/vaultwarden/assets/152861034/b526613f-a07a-4844-a8d7-3704aa025fb4)

**Reply by BlackDex** [reply]
You need to check what happens at the client side. Die the browser report issues, are there any logs in Vaultwarden during these attempts which could explain something. Are there any issues related in the `/admin/diagnostics` page?

**Reply by SagarKandel** [reply]
Nothing happens on the client side. When you say any loss, what/how do you mean?


I don't think so. I have attached the diagnostic page screenshot.
<img width="1462" alt="image" src="https://github.com/dani-garcia/vaultwarden/assets/50504908/2d6719d2-9742-4a71-98e2-93ea92176e7f">

**Reply by BlackDex** [reply]
I mean logs, during the time a user tries to use the invite.

**Reply by xewl** [reply]
I had the same on the latest docker tag. The invite goes through, the user gets created and has its own vault; but it user never moves over to "needs confirmation" and has no org linked (not in admin either), after creating the account. Reverted back to 1.29.2, reinvited and that worked (had a notif after registration, and now has access to the org as given).

**Reply by BlackDex** [reply]
@xewl same for you, what do the Vaultwarden logs repor?

**Reply by BlackDex** [reply]
I just tested this with the `testing` tagged version which uses `2024.3.0` as a web-vault, and it works just fine for me.
So, there must be something on either the client side, which might change the URI, or the web-vault is cached and the new one is not loaded fully.

**Reply by xewl** [reply]
I'll try to reproduce this later.

**Reply by SagarKandel** [reply]
Here's  log file. 
This should tell more info, maybe.  

[vaultwarden (1).log](https://github.com/dani-garcia/vaultwarden/files/15103044/vaultwarden.1.log)

**Reply by BlackDex** [reply]
I do see the correct call to the invite endpoint, and it returns a 200.
So, that should be a valid token, and valid other stuff.

So not sure why it then is not working.
Please let the user try to logout first and then use the token.
Also, send a new invite just to be sure.

**Reply by SagarKandel** [reply]
I did try doing those steps already. Is there something I need to be wary of to get this working? vault warden is behind the reverse proxy (Nginx)

**Reply by BlackDex** [reply]
i would suggest to try the `testing` tagged container image.

**Reply by SagarKandel** [reply]
yes sure , i will do that

**Reply by SagarKandel** [reply]
testing tag image works, so there must be something within the latest tag image.

**Reply by BlackDex** [reply]
Yes, the web-vault had some issues. But that wad only if there was lock in the username/email if i remember correctly.

**Reply by siriusgithub** [reply]
Had this issue yesterday, if 2FA is req for the org, have the person set that up (if it doesn't give them a message to do so on acct creation) and then log out and then do the email join link again. That somehow triggered the confirmation on the admin site to approve.

**Reply by jobritz** [reply]
Maybe take a look at this discussion? #4157 I think you might have the same problem

**Reply by pablog12** [reply]
I have this same problem and my Domain is set correctly. 

I tried deleting the user and recreating it. That worked for that particular user, but it happened with 5 other users aswell.
All the config seems fine like the ones shared on this thread.

So any idea what might be it?
